odml/property.py

```diff
@@ -102,7 +102,7 @@ def parent(self):
     @parent.setter
     def parent(self, new_parent):
         if new_parent is None:
-            return
+            self._parent = None
         elif self._validate_parent(new_parent):
```

Comment:
I added a removal of a child from parent's `_props` list, after assigning to the child a new parent. @jgrewe Correct me if it should be done in another way.